### DISCUSSION TITLE
null-ls, goimports-reviser issue on Ctrl-S, :w

### DISCUSSION THREAD

**Discussion by kcaashish** [LazyVim/LazyVim#484] [General]
I don't understand if this is an issue with `null-ls` or `goimport-reviser` so I am asking here to confirm first it is not something to do with LazyVim.

My config for `goimports-reviser` is:
```lua
return {
  {
    "jose-elias-alvarez/null-ls.nvim",
    opts = function(_, opts)
      local nls = require("null-ls")
      local nls_sources = {
        -- go
        nls.builtins.formatting.goimports_reviser.with({
          generator_opts = {
            command = "goimports-reviser",
            args = { "-output", "stdout", "$FILENAME" },
            to_stdin = true,
          },
        }),
      }
      vim.list_extend(opts.sources, nls_sources, 1, #nls_sources)
    end,
  },
}
```

In a very simple go project with `go.mod`:
```gomod
module minimal

go 1.19
```

`main.go`:
```go
package main

func main() {
	fmt.Println("Hello, World!")
}
```
Notice the absence of imports statment as well. So there should be a `undeclared name: fmt` erro.

If I delete the line fmt.Println() `dd` then hit `Ctrl-S`, the line pops back up, same with `:w`, doesn't import. How can I look for a solution? Is it an issue with `goimports-reviser`?

**Comment by folke** [comment]
I have no idea. Does that goimports-reviser work with null-ls? You should ask there

**Comment by cbrake** [comment]
I managed to get imports working with gopls, but this obviously is not the reviser: https://github.com/LazyVim/LazyVim/discussions/682#discussioncomment-5861853
